airflow/providers/google/marketing_platform/hooks/search_ads.py

```diff
@@ -19,73 +19,154 @@
+        api_version: str | None = None,
         gcp_conn_id: str = "google_cloud_default",
-        delegate_to: str | None = None,
-        impersonation_chain: str | Sequence[str] | None = None,
     ) -> None:
         super().__init__(
             gcp_conn_id=gcp_conn_id,
-            delegate_to=delegate_to,
-            impersonation_chain=impersonation_chain,
         )
-        self.api_version = api_version
+        self.api_version = api_version or self.default_api_version
 
-    def get_conn(self):
-        """Retrieve connection to Google SearchAds."""
+    def get_conn(self) -> Any:
         if not self._conn:
             http_authorized = self._authorize()
             self._conn = build(
-                "doubleclicksearch",
+                "searchads360",
                 self.api_version,
                 http=http_authorized,
                 cache_discovery=False,
             )
         return self._conn
 
-    def insert_report(self, report: dict[str, Any]) -> Any:
+    @property
+    def customer_service(self):
+        return self.get_conn().customers()
+
+    @property
+    def fields_service(self):
+        return self.get_conn().searchAds360Fields()
+
+    def search(
+        self,
+        customer_id: str,
+        query: str,
+        page_token: str | None = None,
+        page_size: int = 10000,
+        return_total_results_count: bool = False,
+        summary_row_setting: str | None = None,
+        validate_only: bool = False,
+    ):
         """
-        Insert a report request into the reporting system.
+        Search and download the report. Use pagination to download entire report.
 
-        :param report: Report to be generated.
+        :param customer_id: The ID of the customer being queried.
+        :param query: The query to execute.
+        :param page_token: Token of the page to retrieve. If not specified, the first page of results will be
+                           returned. Use the value obtained from `next_page_token` in the previous response
+                           in order to request the next page of results.
+        :param page_size: Number of elements to retrieve in a single page. When too large a page is requested,
+                          the server may decide to further limit the number of returned resources.
+                          Default is 10000.
+        :param return_total_results_count: If true, the total number of results that match the query ignoring
+                          the LIMIT clause will be included in the response. Default is false.
+        :param summary_row_setting: Determines whether a summary row will be returned. By default,
+                          summary row is not returned. If requested, the summary row will be sent
+                          in a response by itself after all others query results are returned.
+        :param validate_only: If true, the request is validated but not executed. Default is false.
```

Comment:
Is it correct indentation?

Reply:
Ohh, I was trying to match the previous line but it seems we are using 4 space indentations for continuing lines